dozer-cli/src/simple/cloud/login.rs

```diff
@@ -0,0 +1,170 @@
+        fs::create_dir_all(CredentialInfo::get_directory_path())
+            .map_err(CloudCredentialError::FailedToCreateDirectory)?;
+        let f = std::fs::OpenOptions::new()
+            .create(true)
+            .write(true)
+            .open(file_path)?;
+        let mut current_credential_infos: Vec<CredentialInfo> = CredentialInfo::read_profile()?;
+        current_credential_infos.append(&mut vec![self.clone()]);
+        current_credential_infos.dedup_by_key(|key| key.to_owned().profile_name);
+        serde_yaml::to_writer(f, &current_credential_infos)?;
+        Ok(())
+    }
+
+    fn read_profile() -> Result<Vec<CredentialInfo>, CloudCredentialError> {
+        let file_path = CredentialInfo::get_file_path();
+        let file = std::fs::File::open(file_path)
+            .map_err(|_e| CloudCredentialError::MissingCredentialFile)?;
+        let credential_info: Vec<CredentialInfo> =
+            serde_yaml::from_reader(file).map_err(CloudCredentialError::SerializationError)?;
+        Ok(credential_info)
+    }
+
+    pub fn load(name: Option<String>) -> Result<CredentialInfo, CloudCredentialError> {
+        let credential_info: Vec<CredentialInfo> = CredentialInfo::read_profile()?;
+        if let Some(name) = name {
+            let credential_info = credential_info
+                .into_iter()
+                .find(|info| info.profile_name == name)
+                .ok_or(CloudCredentialError::MissingProfile)?;
+            return Ok(credential_info);
+        }
+        let loaded_profile = credential_info
+            .into_iter()
+            .next()
+            .ok_or(CloudCredentialError::MissingProfile)?;
+        Ok(loaded_profile)
+    }
+
+    pub async fn get_access_token(&self) -> Result<TokenResponse, CloudCredentialError> {
+        let client = reqwest::Client::builder()
+            .build()
+            .map_err(CloudCredentialError::HttpRequestError)?;
+        let mut headers = reqwest::header::HeaderMap::new();
+        headers.insert(
+            "Content-Type",
+            "application/x-www-form-urlencoded".parse().unwrap(),
+        );
+
+        let mut params: HashMap<&str, &str> = HashMap::new();
+        params.insert("grant_type", "client_credentials");
+        params.insert("client_id", self.client_id.as_str());
+        params.insert("client_secret", self.client_secret.as_str());
+        let request = client
+            .request(reqwest::Method::POST, self.auth_url.to_owned())
+            .headers(headers)
+            .form(&params);
+        let response = request
+            .send()
+            .await
+            .map_err(CloudCredentialError::HttpRequestError)?;
+        let json_response: Value = response
+            .json()
+            .await
+            .map_err(CloudCredentialError::HttpRequestError)?;
+        let result: TokenResponse = serde_json::from_value(json_response)
+            .map_err(CloudCredentialError::JsonSerializationError)?;
+        Ok(result)
```

Comment:
Same here, do you need to use this instead of just returning from previous line?